source/common/http/http2/codec_impl.cc

```diff
@@ -538,18 +546,23 @@ int ConnectionImpl::onInvalidFrame(int32_t stream_id, int error_code) {
   ENVOY_CONN_LOG(debug, "invalid frame: {} on stream {}", connection_, nghttp2_strerror(error_code),
                  stream_id);
 
+  StreamImpl* stream = getStream(stream_id);
   // The stream is about to be closed due to an invalid header or messaging. Don't kill the
   // entire connection if one stream has bad headers or messaging.
   if (error_code == NGHTTP2_ERR_HTTP_HEADER || error_code == NGHTTP2_ERR_HTTP_MESSAGING) {
     stats_.rx_messaging_error_.inc();
-    StreamImpl* stream = getStream(stream_id);
     if (stream != nullptr) {
       // See comment below in onStreamClose() for why we do this.
       stream->reset_due_to_messaging_error_ = true;
     }
     return 0;
   }
 
+  if (stream != nullptr) {
+    ENVOY_CONN_LOG(warn, "Referring the stream later may cause crash. See #6842. id: {}, error: {}",
```

Comment:
I don't think I would do a warn here. If you can't test this, let's just do `ASSERT(stream != nullptr);` ?

Reply:
I think the stream can have a value.  If after onInvalidFrame() returns error, the connection is closed, or stops processing, the crash won't happen.

Reply:
But then I don't think we should warn here if this can happen? This could spam the lags and I'm not sure who would look at this and how they would action on it?